packages/lodestar/src/chain/factory/block/index.ts

```diff
@@ -22,32 +22,37 @@ export async function assembleBlock(
   randaoReveal: Bytes96,
   graffiti = ZERO_HASH,
 ): Promise<BeaconBlock | null> {
-  const [parentBlock, currentState] = await Promise.all([
+  const [parentBlock, stateContext] = await Promise.all([
     chain.getHeadBlock(),
-    chain.getHeadState()
+    chain.getHeadContext()
```

Comment:
If we keep the `getHeadState`, we can avoid a lot of these changes

Reply:
TBH, I initially removed so I'm forced to fix bugs where we use head state incorectly (that's how I've discovered bugs in gossip validation). I will restore getHeadState method